sphinx/themes/basic/static/searchtools.js

```diff
@@ -160,10 +160,15 @@ const Search = {
   _queued_query: null,
   _pulse_status: -1,
 
-  htmlToText: (htmlString) => {
+  htmlToText: (htmlString, anchor) => {
     const htmlElement = new DOMParser().parseFromString(htmlString, 'text/html');
     htmlElement.querySelectorAll(".headerlink").forEach((el) => { el.remove() });
-    const docContent = htmlElement.querySelector('[role="main"]');
+    let docContent = undefined;
+    if (anchor && htmlElement.querySelector(anchor)) {
+      docContent = htmlElement.querySelector(anchor);
```

Comment:
I have a concern here: I think this is going to retrieve the text content within the anchor element?

In practice we may want to retrieve some of the following HTML sibling element text as well.

We do have some [test coverage on `searchtools.js`](https://github.com/sphinx-doc/sphinx/blob/ceb3b2a904cb14f959917873e8be91fb194ca393/tests/js/searchtools.js) and I wonder whether we could use that to make some assertions about what we retrieve and summarize (and also safeguard the feature against regressions).

Reply:
I don't think we want the sibling element. The anchor defines a section within the documentation. I don't think we want (or need) any text outside of it: it's (1) likely to be out of context and (2) probably outside of the [text we'd take for the snippit](https://github.com/wlach/sphinx/blob/27f4d3605989ea2216c5f370b07833a501de7267/sphinx/themes/basic/static/searchtools.js#L563) except in edge cases.

Here's the section for the 2to3 example you gave earlier:

![image](https://github.com/sphinx-doc/sphinx/assets/20569/1360e661-870f-4428-aa39-1bb4e8fd4978)

Reply:
Ok, brilliant.  I'm doing a few more tests locally, and getting around to agreeing with all this.